keps/sig-auth/3766-referencegrant/README.md

```diff
@@ -441,174 +368,365 @@ rejected".
+  Resource string `json:"resource"`
+}
 
-Comments from @thockin:
+// Namespace describes trusted namespaces. Exactly one of Name or Selector MUST be specified.
+// +oneOf
+type ReferenceGrantNamespace struct {
+  // Name is the name of the namespace.
+  Name string `json:"name"`
 
-I would not argue for subject/object - that's confusing, and I love to take
-analogies too far.
+  // Selector selects namespaces based on their labels.
+  Selector metav1.LabelSelector `json:"selector"`
+}
+
+
+// ReferenceGrantTo describes what resources are allowed as targets of the
+// references.
+type ReferenceGrantTo struct {
+  // Group is the group of the referent.
+  // When empty, the Kubernetes core API group is inferred.
+  Group string `json:"group"`
+
+  // Resource is the resource of the referent.
+  Resource string `json:"resource"`
+
+  // Name is the name of the referent. When unspecified, this policy
+  // refers to all resources of the specified Group and Kind in the local
+  // namespace.
+  //
+  // +optional
+  Name *string `json:"name,omitempty"`
+}
+```
 
-NetPol uses To/From but only for actual communications, and that still has been
-critiqued as perhaps "too cute".
+### Variations from Gateway API ReferenceGrant
 
-Subject/From isn't too bad, but not as symmetric. Subject/Referrer is correct
-but decidedly uncute. Subject/Origin?
+This KEP makes several changes from the existing ReferenceGrant resource
+defined by Gateway API:
 
-I hold opinions from an API review POV, but I'd like sig-auth to own the
-decision :)
+#### 1. Removing Space for Status
 
-For reference, there was an [earlier
-discussion](https://groups.google.com/g/kubernetes-api-reviewers/c/ldmrXXQC4G4)
-on the kubernetes-api-reviewers mailing list that's also relevant to this.
-<<[/UNRESOLVED]>>
+Although we had theoretical use cases for status, they never materialized into
+the Gateway API version of ReferenceGrant. [Earlier feedback on this
+KEP](https://github.com/kubernetes/enhancements/pull/3767#discussion_r1084670421)
+showed that we could use alternative approaches such as Events or new
+"RefGranted" resources published by implementing controllers. Given that there
+are both reasonable alternatives to status and we have not needed it so far, it
+seems reasonable to leave out of this version of the API.
 
-```go
+This means that `Spec` has been removed from the API, and `Spec.To` and
+`Spec.From` have become top-level `To` and `From` fields.
 
-// ReferenceGrantFrom describes trusted namespaces and kinds.
-type ReferenceGrantFrom struct {
-	// Group is the group of the referent.
-	// When empty, the Kubernetes core API group is inferred.
-	Group Group `json:"group"`
+#### 2. Kind -> Resource
 
-	// Resource is the resource of the referent.
-	Resource string `json:"resource"`
+In the original Gateway API implementation, we chose to use `Kind` rather than
+`Resource`, mainly to improve the user experience. That is, it's easier users
```

Comment:
```suggestion
In the original Gateway API implementation, we chose to use `kind` rather than
`resource`, mainly to improve the user experience. That is, it's easier for users
```